test/python/transpiler/test_linear_functions_passes.py

```diff
@@ -329,6 +329,53 @@ def test_connected_gates(self):
         self.assertNotIn("cx", circuit2.count_ops().keys())
         self.assertNotIn("swap", circuit2.count_ops().keys())
 
+    def test_if_else(self):
+        """Test that collection recurses into a simple if-else."""
+        pass_ = CollectLinearFunctions()
+
+        circuit = QuantumCircuit(4)
+        circuit.cx(0, 1)
+        circuit.cx(1, 0)
+        circuit.cx(2, 3)
+
+        test = QuantumCircuit(4, 1)
+        test.h(0)
+        test.measure(0, 0)
+        test.if_else((0, True), circuit.copy(), circuit.copy(), range(4), [0])
+
+        expected = QuantumCircuit(4, 1)
+        expected.h(0)
+        expected.measure(0, 0)
+        expected.if_else((0, True), pass_(circuit), pass_(circuit), range(4), [0])
```

Comment:
It seems like this pattern of creating the expected control flow blocks by calling the pass on them when creating the control flow instruction could be repeated in some of the other tests e.g. `test_optimize_swap_before_measure`, `test_cx_cancellation`, `test_clifford_passes`, etc.

Reply:
I actually swapped from this style to the pure style about halfway through - I thought being _completely_ explicit about the expected form might be neater. I'm happy to go either way - which do you prefer?

Reply:
I used explicit in my tests but I thought this was a nice approach. One has to be a little careful with nested statements but perhaps no more than being explicit.

Reply:
I approved since we can perhaps worry about it later if it comes up.